src/connectivity/mod.rs

```diff
@@ -664,6 +664,67 @@ pub fn digraph_core_number(
     core_number::core_number(py, &graph.graph)
 }
 
+/// Return the articulation points of an undirected graph.
+///
+/// An articulation point or cut vertex is any node whose removal (along with
+/// all its incident edges) increases the number of connected components of
+/// a graph. An undirected connected graph without articulation points is
+/// biconnected.
+///
+/// .. note::
+///
+///     The function implicitly assumes that there are no parallel edges
+///     or self loops. It may produce incorrect/unexpected results if the
+///     input graph has self loops or parallel edges. It's also a recursive
+///     implementation and might run out of memory in large graphs.
+///
+/// :param PyGraph: The undirected graph to be used.
+///
+/// :returns: A set with node indexes of the articulation points in the graph.
+/// :rtype: set
+#[pyfunction]
+#[pyo3(text_signature = "(graph, /)")]
+pub fn articulation_points(graph: &graph::PyGraph) -> HashSet<usize> {
```

Comment:
The output set size depends on the graph but I don't expect it to be significantly large in most cases, so using a `HashSet` instead of a custom return type should be fine.